### PR TITLE
add quickbar proxy

### LINKED ISSUE #235: quickbar proxy
to avoid external domains, we need a redirect for https://arrrg.de/serlo-stats/quickbar.json, e.g. from https://de.serlo.org/api/stats/quickbar.json

### PR DESCRIPTION
**Pull request by elbotho**
closes #235

### CODE REVIEW

File: src/quickbarProxy.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import { Url } from './utils'
+
+export async function quickbarProxy(
+  request: Request
+): Promise<Response | null> {
+  // https://de.serlo.org/api/stats/quickbar.json
+  const url = Url.fromRequest(request)
+
+  if (url.pathname !== '/api/stats/quickbar.json') return null
```
Comment (elbotho) [inline src/quickbarProxy.ts:31]: only needs to support `DE` for now, that's why I don't check the subdomain here.

File: __tests__/quickbarProxy.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/**
```
Comment (kulla) [inline __tests__/quickbarProxy.ts:1]: rename to `__tests__/quickbar-proxy.ts`

File: __tests__/quickbarProxy.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+import { RestResolver, currentTestEnvironment } from './__utils__'
+
+describe('de.serlo.org/api/stats/quickbar.json', () => {
+  beforeEach(() => {
+    givenArrrg(arrrgServer())
+  })
+
+  test('returns json file', async () => {
+    const response = await requestJson('/api/stats/quickbar.json')
```
Comment (kulla) [inline __tests__/quickbarProxy.ts:32]:
```suggestion
   const env = currentTestEnvironment()
    const response = await env.fetch({ pathname: '/api/stats/quickbar.json' })
```

File: __tests__/quickbarProxy.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  function arrrgServer(): RestResolver {
+    return (req, res, ctx) => {
+      return res(
+        ctx.set('content-type', 'application/json'),
+        ctx.body(
+          '[{"id":"19767","title":"Mathematik Startseite","path":[],"isTax":false,"count":35416}]'
+        )
+      )
+    }
+  }
```
Comment (kulla) [inline __tests__/quickbarProxy.ts:56]: Can be merged together in one function like `setupQuickbarServer()`.

File: src/quickbarProxy.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+import { Url } from './utils'
+
+export async function quickbarProxy(
+  request: Request
+): Promise<Response | null> {
+  // https://de.serlo.org/api/stats/quickbar.json
+  const url = Url.fromRequest(request)
+
```
Comment (kulla) [inline src/quickbarProxy.ts:30]:
```suggestion
if (!isInstance(url.subdomain)) return null
```

File: src/quickbarProxy.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  if (url.pathname !== '/api/stats/quickbar.json') return null
+
+  try {
+    const jsonUrl = 'https://arrrg.de/serlo-stats/quickbar.json'
+    const jsonRes = await fetch(jsonUrl, {
+      cf: { cacheTtl: 24 * 60 * 60 },
+    } as unknown as RequestInit)
+    if (jsonRes.ok) return jsonRes
+  } catch (e) {
+    // this should be reported, right?
```
Comment (kulla) [inline src/quickbarProxy.ts:40]: See https://github.com/serlo/serlo.org-cloudflare-worker/blob/main/src/metadata-api.ts for a good example. You may need to add a `SentryReporter.captureException()` method like https://github.com/serlo/serlo.org-cloudflare-worker/blob/main/src/utils/sentry.ts#L44-L46 See https://github.com/robertcepa/toucan-js#features for the methods the object returned by `getToucan()` supports..

**Review by kulla — CHANGES_REQUESTED** [review]

**Review by kulla — APPROVED** [review]

**Pull request merged**